File: python-village/INI2_Variables_and_Some_Arithmetic/variAndArith.pl
#!/usr/bin/perl

# Program 'variAndArith'
# Given: Two positive integers a and b, each less than 1000
# Return: The integer corresponding to the square of the hypotenuse of
#     the right triangle whose legs have lengths a and b
# Date: 2015 Sept 15
# Version: 5.16

use strict;
use warnings;

open( IN1, $ARGV[0] ) || die "input problem";

my @array = split " ", <IN1>;
our $total; # Define the square of the hypotenuse

foreach ( @array ){
    $total += $_ ** 2; # Add sqaure of each leg
}

open( MYFILE, '>out.txt' );
print MYFILE $total, "\n";
close( MYFILE );

File: python-village/INI2_Variables_and_Some_Arithmetic/variAndArith.scala
// Program: 'variAndArith.scala'
// Given: Two positive integers a and b, each less than 1000
// Return: The integer corresponding to the square of the hypotenuse of the
//     right triangle whose legs have lengths a and b
// Date: 2015 Oct 14
// Version: 2.11.7

import io.Source

// read data in as array
val source = Source.fromFile("rosalind_ini2.txt")
val lines = source.getLines.next.split(" ")

var total = 0;

// loop through array and add to total
for (n <- lines) total += math.pow(n.toInt, 2).toInt

// write answer to output file
scala.tools.nsc.io.File("out.txt").writeAll(total.toString)
